### PR TITLE
Pass along enhanced task events.

### PR DESCRIPTION
**Pull request by katrogan**
# TL;DR

Adds the following to TaskExecutionMetadata
- reason
-  task type
- resource pool (flyte resource manager) information
- generated task name

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/325

## Follow-up issue
_NA_

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -6,8 +6,8 @@ require (
 	github.com/DiSiqueira/GoTree v1.0.1-0.20180907134536-53a8e837f295
 	github.com/benlaurie/objecthash v0.0.0-20180202135721-d1e3d6079fc1
 	github.com/fatih/color v1.10.0
-	github.com/flyteorg/flyteidl v0.18.20
-	github.com/flyteorg/flyteplugins v0.5.38
+	github.com/flyteorg/flyteidl v0.18.25
+	github.com/flyteorg/flyteplugins v0.5.39-0.20210322202107-51b9713edaba
```
Comment (katrogan) [inline go.mod:10]: will update before merging

File: pkg/controller/composite_workqueue.go [hunk 2]
```diff
@@ -14,9 +14,9 @@ import (
 )
 
 // A CompositeWorkQueue can be used in cases where the work is enqueued by two sources. It can be enqueued by either
-// 1. Informer for the Primary Object itself. In case of FlytePropeller, this is the workflow object
+// 1. Informer for the Primary Object itself. InputReader case of FlytePropeller, this is the workflow object
```
Comment (EngHabu) [inline pkg/controller/composite_workqueue.go:17]: Unintended renames?

File: pkg/controller/nodes/task/transformer.go [hunk 3]
```diff
@@ -69,55 +69,73 @@ func getParentNodeExecIDForTask(taskExecID *core.TaskExecutionIdentifier, execCo
...
 			return nil, err
 		}
 	}
 
-	nodeExecutionID, err := getParentNodeExecIDForTask(taskExecID, execContext)
+	nodeExecutionID, err := getParentNodeExecIDForTask(input.TaskExecID, input.ExecContext)
 	if err != nil {
 		return nil, err
 	}
+	metadata := input.Info.Info().Metadata
```
Comment (EngHabu) [inline pkg/controller/nodes/task/transformer.go:99]: We should populate Metadata with all the things we know/control in propeller... GeneratedName, ResourcePoolInfo... (etc.?)

File: pkg/controller/composite_workqueue.go (hunk 2 above)
Comment (katrogan) [reply pkg/controller/composite_workqueue.go:17]: goland you're 2 clever

File: pkg/controller/nodes/task/transformer.go (hunk 3 above)
Comment (katrogan) [reply pkg/controller/nodes/task/transformer.go:100]: done, PTAL

File: go.mod [hunk 4]
```diff
@@ -6,8 +6,8 @@ require (
 	github.com/DiSiqueira/GoTree v1.0.1-0.20180907134536-53a8e837f295
 	github.com/benlaurie/objecthash v0.0.0-20180202135721-d1e3d6079fc1
 	github.com/fatih/color v1.10.0
-	github.com/flyteorg/flyteidl v0.18.24
-	github.com/flyteorg/flyteplugins v0.5.38
+	github.com/flyteorg/flyteidl v0.18.25
+	github.com/flyteorg/flyteplugins v0.5.39-0.20210323171635-da37ae79061e
```
Comment (katrogan) [inline go.mod:10]: will update before merging

**Comment by katrogan** [comment]
PTAL @EngHabu @kumare3

**Comment by katrogan** [comment]
@EngHabu mind taking another look?

**Review by EngHabu — APPROVED** [review]

**Pull request merged**